aqt/archives.py

```diff
@@ -17,19 +17,24 @@
 # IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
 # FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
 # COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
-# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR I    packagingN
 # CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 import xml.etree.ElementTree as ElementTree
 from logging import getLogger
 
+from semantic_version import Version
+
 from aqt.exceptions import ArchiveListError, NoPackageFound
 from aqt.helper import Settings, getUrl
 
 
 class TargetConfig:
     def __init__(self, version, target, arch, os_name):
-        self.version = version
+        if isinstance(version, Version):
+            self.version = version.public
```

Comment:
`version.public` is a data member of `packaging.version.Version`, but not `semantic_version.Version`.

If you need to turn a `Version` into a string, you can always use `str(version)`.  Both version classes provide a `.__str__()` method.